catalogs/catalog-jdbc-starrocks/src/main/java/org/apache/gravitino/catalog/starrocks/operations/StarRocksTablePartitionOperations.java

```diff
@@ -53,26 +78,162 @@ public StarRocksTablePartitionOperations(
+    try (Connection connection = getConnection(loadedTable.databaseName())) {
+      Transform partitionInfo = loadedTable.partitioning()[0];
+      Map<String, Type> columnTypes = getColumnType(connection);
+      String showPartitionsSql = String.format("SHOW PARTITIONS FROM `%s`", loadedTable.name());
+      try (Statement statement = connection.createStatement();
+          ResultSet result = statement.executeQuery(showPartitionsSql)) {
+        ImmutableList.Builder<Partition> partitions = ImmutableList.builder();
+        while (result.next()) {
+          partitions.add(
+              StarRocksUtils.fromStarRocksPartition(
+                  loadedTable.name(), result, partitionInfo, columnTypes));
+        }
+        return partitions.build().toArray(new Partition[0]);
+      }
+    } catch (SQLException e) {
+      throw exceptionConverter.toGravitinoException(e);
+    }
   }
 
   @Override
   public Partition getPartition(String partitionName) throws NoSuchPartitionException {
-    throw new NotImplementedException("To be implemented in the future");
+    try (Connection connection = getConnection(loadedTable.databaseName())) {
+      Transform partitionInfo = loadedTable.partitioning()[0];
+      Map<String, Type> columnTypes = getColumnType(connection);
+      String showPartitionsSql =
+          String.format(
+              "SHOW PARTITIONS FROM `%s` WHERE PartitionName = \"%s\"",
+              loadedTable.name(), partitionName);
+      try (Statement statement = connection.createStatement();
+          ResultSet result = statement.executeQuery(showPartitionsSql)) {
+        if (result.next()) {
+          return StarRocksUtils.fromStarRocksPartition(
+              loadedTable.name(), result, partitionInfo, columnTypes);
+        }
+      }
+    } catch (SQLException e) {
+      throw exceptionConverter.toGravitinoException(e);
+    }
+    throw new NoSuchPartitionException("Partition %s does not exist", partitionName);
   }
 
   @Override
   public Partition addPartition(Partition partition) throws PartitionAlreadyExistsException {
-    throw new NotImplementedException("To be implemented in the future");
+    try (Connection connection = getConnection(loadedTable.databaseName())) {
+      Transform partitionInfo = loadedTable.partitioning()[0];
+
+      String addPartitionSqlFormat = "ALTER TABLE `%s` ADD %s";
+      String partitionSqlFragment;
+      Partition added;
+
+      if (partition instanceof RangePartition) {
+        Preconditions.checkArgument(
+            partitionInfo instanceof Transforms.RangeTransform,
+            "Table %s is non-range-partitioned, but trying to add a range partition",
+            loadedTable.name());
+
+        RangePartition rangePartition = (RangePartition) partition;
+        partitionSqlFragment = StarRocksUtils.generatePartitionSqlFragment(rangePartition);
+
+        // The partition properties actually cannot be passed into Doris, we just return an empty
```

Comment:
Doris -> StarRocks, also please check the same problem in L65 of `StarRocksColumnDefaultValueConverter`.

Reply:
Done

Reply:
@liunaijie Good, do you have time to work on related documents? You can refer to the document at https://gravitino.apache.org/docs/0.9.1/jdbc-doris-catalog if you are willing to do it, thanks.

Reply:
Sure, I will add the document later

Reply:
The E2E test and Document are in this pr https://github.com/apache/gravitino/pull/7792/commits

Reply:
I see.